# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: media/libcubeb/src/cubeb_resampler.cpp
# Commit: f980191897a2
# Full Hash: f980191897a2ceb0514f83df997dc05a349f54db
# Author: Alex Chronopoulos <achronop@gmail.com>
# Date: 2020-01-15 09:38:07
# Description:
#   Bug 1604117 - Backout cubeb commit aa63601. r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D59674
# ==============================================================================

diff -r 2ee6c4c25052 -r f980191897a2 media/libcubeb/src/cubeb_resampler.cpp
--- a/media/libcubeb/src/cubeb_resampler.cpp	Wed Jan 15 01:27:10 2020 +0200
+++ b/media/libcubeb/src/cubeb_resampler.cpp	Mon Jan 13 13:09:39 2020 +0000
@@ -66,41 +66,19 @@
          (output_buffer && !input_buffer && (!input_frames_count || *input_frames_count == 0)) ||
          (input_buffer && !output_buffer && output_frames == 0));
 
-  // When we have no pending input data and exactly as much input
-  // as output data, we don't need to copy it into the internal buffer
-  // and can directly forward it to the callback.
-  void * in_buf = input_buffer;
-  unsigned long pop_input_count = 0u;
-  if (input_buffer && !output_buffer) {
+  if (input_buffer) {
+    if (!output_buffer) {
       output_frames = *input_frames_count;
-  } else if(input_buffer) {
-    if (internal_input_buffer.length() != 0) {
-      // In this case we have pending input data left and have
-      // to first append the input so we can pass it as one pointer
-      // to the callback
-      internal_input_buffer.push(static_cast<T*>(input_buffer),
-                                 frames_to_samples(*input_frames_count));
-      in_buf = internal_input_buffer.data();
-      pop_input_count = frames_to_samples(output_frames);
-    } else if(*input_frames_count > output_frames) {
-      // In this case we have more input that we need output and
-      // fill the overflowing input into internal_input_buffer
-      // Since we have no other pending data, we can nonetheless
-      // pass the current input data directly to the callback
-      assert(pop_input_count == 0);
-      unsigned long samples_off = frames_to_samples(output_frames);
-      internal_input_buffer.push(static_cast<T*>(input_buffer) + samples_off,
-                                 frames_to_samples(*input_frames_count - output_frames));
     }
+    internal_input_buffer.push(static_cast<T*>(input_buffer),
+                               frames_to_samples(*input_frames_count));
   }
 
-  long rv = data_callback(stream, user_ptr, in_buf, output_buffer, output_frames);
+  long rv = data_callback(stream, user_ptr, internal_input_buffer.data(),
+                          output_buffer, output_frames);
 
   if (input_buffer) {
-    if (pop_input_count) {
-      internal_input_buffer.pop(nullptr, pop_input_count);
-    }
-
+    internal_input_buffer.pop(nullptr, frames_to_samples(output_frames));
     *input_frames_count = output_frames;
     drop_audio_if_needed();
   }